aten/src/ATen/native/cuda/TensorModeKernel.cpp

```diff
@@ -0,0 +1,101 @@
+#include <ATen/native/cuda/TensorModeKernel.h>
+#include <ATen/Functions.h>
+#include <ATen/cuda/CUDAConfig.h>
+#include <ATen/native/CanUse32BitIndexMath.h>
+#include <ATen/native/ReduceOpsUtils.h>
+#include <ATen/native/Resize.h>
+#include <ATen/native/TensorCompare.h>
+
+constexpr int MAX_BLOCK_SIZE = AT_ROCM_ENABLED() ? 256 : 1024;
+
+// Maximum size per grid dimension that we assume (compute capability >= 2.0)
+constexpr int64_t MAX_GRID_SIZE = 65535LL;
+
+namespace at {
+namespace native {
+
+void mode_kernel_impl(
+    Tensor& values,
+    Tensor& indices,
+    const Tensor& self,
+    int64_t dim,
+    bool keepdim) {
+  auto self_sizes = ensure_nonempty_vec(self.sizes().vec());
+  int64_t ndim = ensure_nonempty_dim(self.dim());
+  int64_t slice_size = ensure_nonempty_size(self, dim);
+  int64_t slices = self.numel() / slice_size;
+
+  // Resize output value, index Tensors to appropriate sizes (i.e. the same as
+  // the input Tensor, except at dim=dimension, the size is 1)
+  self_sizes[dim] = 1;
```

Comment:
`assert(dim < self_sizes.size());`

Reply:
It appears that `TORCH_INTERNAL_ASSERT_DEBUG_ONLY` does not suppress the linter, but `assert` does.

Reply:
Also this flags a sign-compare problem now, of course. so after the comparison to 0 add `static_cast<std::size_t>(dim)` in the comparison against size.

Reply:
Alright, changed to `assert(0 <= dim && static_cast<size_t>(dim) < self_sizes.size())`